Synthesis failed for nodejs-pubsub

Hello! Autosynth couldn't regenerate nodejs-pubsub. :broken_heart:

Here's the output from running `synth.py`:

```
in-event <InotifyEvent: src_path=b'./node_modules/protobufjs/cli/node_modules/optionator', wd=2498, mask=IN_ISDIR|IN_MOVED_TO, cookie=252401, name=optionator>
DEBUG:watchdog.observers.inotify_buffer:could not find matching move_from event
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/protobufjs/cli/node_modules/source-map', wd=2498, mask=IN_ISDIR|IN_MOVED_TO, cookie=252402, name=source-map>
DEBUG:watchdog.observers.inotify_buffer:could not find matching move_from event
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/protobufjs/cli/node_modules/escodegen', wd=2498, mask=IN_ISDIR|IN_MOVED_FROM, cookie=252403, name=escodegen>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/protobufjs/cli/node_modules/.escodegen.DELETE', wd=2498, mask=IN_ISDIR|IN_MOVED_TO, cookie=252403, name=.escodegen.DELETE>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/protobufjs/cli/node_modules/escodegen', wd=2498, mask=IN_ISDIR|IN_MOVED_TO, cookie=252404, name=escodegen>
DEBUG:watchdog.observers.inotify_buffer:could not find matching move_from event
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/protobufjs/cli/node_modules/.escodegen.DELETE', wd=2498, mask=IN_DELETE|IN_ISDIR, cookie=0, name=.escodegen.DELETE>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/protobufjs/cli/node_modules/uglify-js', wd=2498, mask=IN_ISDIR|IN_MOVED_TO, cookie=252406, name=uglify-js>
DEBUG:watchdog.observers.inotify_buffer:could not find matching move_from event
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/protobufjs/cli/node_modules/.staging', wd=2498, mask=IN_DELETE|IN_ISDIR, cookie=0, name=.staging>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/protobufjs/cli/package.json.67189814', wd=1328, mask=IN_CREATE, cookie=0, name=package.json.67189814>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/protobufjs/cli/package.json.67189814', wd=1328, mask=IN_MODIFY, cookie=0, name=package.json.67189814>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/protobufjs/cli/package.json.67189814', wd=1328, mask=IN_ATTRIB, cookie=0, name=package.json.67189814>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/protobufjs/cli/package.json.67189814', wd=1328, mask=IN_MOVED_FROM, cookie=252425, name=package.json.67189814>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/protobufjs/cli/package.json', wd=1328, mask=IN_MOVED_TO, cookie=252425, name=package.json>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/protobufjs/cli/package-lock.json.4207760345', wd=1328, mask=IN_CREATE, cookie=0, name=package-lock.json.4207760345>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/protobufjs/cli/package-lock.json.4207760345', wd=1328, mask=IN_MODIFY, cookie=0, name=package-lock.json.4207760345>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/protobufjs/cli/package-lock.json.4207760345', wd=1328, mask=IN_ATTRIB, cookie=0, name=package-lock.json.4207760345>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/protobufjs/cli/package-lock.json.4207760345', wd=1328, mask=IN_MOVED_FROM, cookie=252426, name=package-lock.json.4207760345>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/protobufjs/cli/package-lock.json', wd=1328, mask=IN_MOVED_TO, cookie=252426, name=package-lock.json>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./protos/protos.json', wd=5, mask=IN_MODIFY, cookie=0, name=protos.json>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./protos/protos.js', wd=5, mask=IN_MODIFY, cookie=0, name=protos.js>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./protos/protos.d.ts', wd=5, mask=IN_MODIFY, cookie=0, name=protos.d.ts>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./protos/protos.d.ts', wd=5, mask=IN_MODIFY, cookie=0, name=protos.d.ts>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./protos/protos.d.ts', wd=5, mask=IN_MODIFY, cookie=0, name=protos.d.ts>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./protos/protos.d.ts', wd=5, mask=IN_MODIFY, cookie=0, name=protos.d.ts>

> @google-cloud/pubsub@2.4.0 precompile /home/kbuilder/.cache/synthtool/nodejs-pubsub
> gts clean

version: 12
Removing build ...

> @google-cloud/pubsub@2.4.0 compile /home/kbuilder/.cache/synthtool/nodejs-pubsub
> tsc -p . && cp -r protos build/

[96msrc/v1/publisher_client.ts[0m:[93m261[0m:[93m9[0m - [91merror[0m[90m TS2532: [0mObject is possibly 'undefined'.

[7m261[0m         this.descriptors.batching[methodName] ||
[7m   [0m [91m        ~~~~~~~~~~~~~~~~~~~~~~~~~[0m


Found 1 error.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/pubsub@2.4.0 compile: `tsc -p . && cp -r protos build/`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/pubsub@2.4.0 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-08-12T19_08_03_818Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/pubsub@2.4.0 prepare: `npm run compile-protos && npm run compile`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/pubsub@2.4.0 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-08-12T19_08_03_889Z-debug.log
2020-08-12 12:08:03,921 synthtool [ERROR] > Failed executing npm install:

None
ERROR:synthtool:Failed executing npm install:

None
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/nodejs-pubsub/synth.py", line 25, in <module>
    node.postprocess_gapic_library()
  File "/tmpfs/src/github/synthtool/synthtool/languages/node.py", line 193, in postprocess_gapic_library
    install(hide_output=hide_output)
  File "/tmpfs/src/github/synthtool/synthtool/languages/node.py", line 167, in install
    shell.run(["npm", "install"], hide_output=hide_output)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['npm', 'install']' returned non-zero exit status 1.
2020-08-12 12:08:03,994 autosynth [ERROR] > Synthesis failed
HEAD is now at 869ecc9 build: stop passing credential file when running gcloud container (#1083)
Switched to branch 'autosynth'
Removing __pycache__/
Removing node_modules/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 690, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 539, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 670, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 375, in synthesize_loop
    has_changes = toolbox.synthesize_version_in_new_branch(synthesizer, youngest)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 273, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/b742586e-df31-4aac-8092-78288e9ea8e7/targets/github%2Fsynthtool;config=default/tests;query=nodejs-pubsub;failed=false).

--- a/.kokoro/release/docs-devsite.sh
+++ b/.kokoro/release/docs-devsite.sh
@@ -36,6 +36,11 @@ NAME=$(cat .repo-metadata.json | json name)
 
 mkdir ./_devsite
 cp ./yaml/$NAME/* ./_devsite
+
+# Delete SharePoint item, see https://github.com/microsoft/rushstack/issues/1229
+sed -i -e '1,3d' ./yaml/toc.yml
+sed -i -e 's/^    //' ./yaml/toc.yml
+
 cp ./yaml/toc.yml ./_devsite/toc.yml
 
 # create docs.metadata, based on package.json and .repo-metadata.json.

--- a/.kokoro/release/publish.cfg
+++ b/.kokoro/release/publish.cfg
@@ -61,7 +61,7 @@ build_file: "nodejs-pubsub/.kokoro/trampoline.sh"
 # Configure the docker image for kokoro-trampoline.
 env_vars: {
     key: "TRAMPOLINE_IMAGE"
-    value: "gcr.io/cloud-devrel-kokoro-resources/node:8-user"
+    value: "gcr.io/cloud-devrel-kokoro-resources/node:12-user"
 }
 
 env_vars: {

--- a/src/v1/publisher_client.ts
+++ b/src/v1/publisher_client.ts
@@ -214,7 +214,7 @@ export class PublisherClient {
     // requests; denote this.
 
     this.descriptors.batching = {
-      Publish: new this._gaxModule.BundleDescriptor(
+      publish: new this._gaxModule.BundleDescriptor(
         'messages',
         ['topic'],
         'message_ids',
@@ -295,12 +295,14 @@ export class PublisherClient {
         }
       );
 
+      const descriptor =
+        this.descriptors.page[methodName] ||
+        this.descriptors.batching?.[methodName] ||
+        undefined;
       const apiCall = this._gaxModule.createApiCall(
         callPromise,
         this._defaults[methodName],
-        this.descriptors.page[methodName] ||
-          this.descriptors.stream[methodName] ||
-          this.descriptors.longrunning[methodName]
+        descriptor
       );
 
       this.innerApiCalls[methodName] = apiCall;

--- a/src/v1/subscriber_client.ts
+++ b/src/v1/subscriber_client.ts
@@ -281,12 +281,14 @@ export class SubscriberClient {
         }
       );
 
+      const descriptor =
+        this.descriptors.page[methodName] ||
+        this.descriptors.stream[methodName] ||
+        undefined;
       const apiCall = this._gaxModule.createApiCall(
         callPromise,
         this._defaults[methodName],
-        this.descriptors.page[methodName] ||
-          this.descriptors.stream[methodName] ||
-          this.descriptors.longrunning[methodName]
+        descriptor
       );
 
       this.innerApiCalls[methodName] = apiCall;

--- a/synth.metadata
+++ b/synth.metadata
@@ -3,23 +3,23 @@
     {
       "git": {
         "name": ".",
-        "remote": "https://github.com/googleapis/nodejs-pubsub.git",
-        "sha": "99ae7fbcb3cb7f1273e68cfa88e855bfce25b76f"
+        "remote": "git@github.com:googleapis/nodejs-pubsub.git",
+        "sha": "869ecc992bbee934ee630e7f38d7e88234c9e80a"
       }
     },
     {
       "git": {
         "name": "googleapis",
         "remote": "https://github.com/googleapis/googleapis.git",
-        "sha": "706053bbfb3f16ef752e513c2898a1f26cdd8e41",
-        "internalRef": "323803770"
+        "sha": "874846a1917ee5c3fe271449f3cb9a06e75407be",
+        "internalRef": "326288259"
       }
     },
     {
       "git": {
         "name": "synthtool",
         "remote": "https://github.com/googleapis/synthtool.git",
-        "sha": "94421c47802f56a44c320257b2b4c190dc7d6b68"
+        "sha": "5747555f7620113d9a2078a48f4c047a99d31b3e"
       }
     }
   ],

